barretenberg/cpp/src/barretenberg/flavor/goblin_ultra.hpp

```diff
@@ -78,137 +80,87 @@ class GoblinUltra {
+                              q_r,             // column 3
+                              q_o,             // column 4
+                              q_4,             // column 5
+                              q_arith,         // column 6
+                              q_sort,          // column 7
+                              q_elliptic,      // column 8
+                              q_aux,           // column 9
+                              q_lookup,        // column 10
+                              q_busread,       // column 11
+                              sigma_1,         // column 12
+                              sigma_2,         // column 13
+                              sigma_3,         // column 14
+                              sigma_4,         // column 15
+                              id_1,            // column 16
+                              id_2,            // column 17
+                              id_3,            // column 18
+                              id_4,            // column 19
+                              table_1,         // column 20
+                              table_2,         // column 21
+                              table_3,         // column 22
+                              table_4,         // column 23
+                              lagrange_first,  // column 24
+                              lagrange_last,   // column 25
+                              lagrange_ecc_op, // column 26 // indicator poly for ecc op gates
+                              databus_id)      // column 27 // id polynomial, i.e. id_i = i
 
         static constexpr CircuitType CIRCUIT_TYPE = CircuitBuilder::CIRCUIT_TYPE;
 
-        std::vector<HandleType> get_selectors() override
+        RefVector<DataType> get_selectors()
         {
             return { q_m, q_c, q_l, q_r, q_o, q_4, q_arith, q_sort, q_elliptic, q_aux, q_lookup, q_busread };
         };
-        std::vector<HandleType> get_sigma_polynomials() override { return { sigma_1, sigma_2, sigma_3, sigma_4 }; };
-        std::vector<HandleType> get_id_polynomials() override { return { id_1, id_2, id_3, id_4 }; };
-
-        std::vector<HandleType> get_table_polynomials() { return { table_1, table_2, table_3, table_4 }; };
+        RefVector<DataType> get_sigma_polynomials() { return { sigma_1, sigma_2, sigma_3, sigma_4 }; };
+        RefVector<DataType> get_id_polynomials() { return { id_1, id_2, id_3, id_4 }; };
+        RefVector<DataType> get_table_polynomials() { return { table_1, table_2, table_3, table_4 }; };
     };
 
     /**
      * @brief Container for all witness polynomials used/constructed by the prover.
      * @details Shifts are not included here since they do not occupy their own memory.
      */
-    template <typename DataType, typename HandleType>
-    class WitnessEntities : public WitnessEntities_<DataType, HandleType, NUM_WITNESS_ENTITIES> {
+    template <typename DataType> class WitnessEntities {
```

Comment:
In the ECCVM flavor, `WitnessEntities` has become `WireEntities` and had its derived witnesses split out. That hasn't happened here. Is that intentional?

Reply:
It was just a small class. I guess I applied 'do stuff to break up the really big flavor entity listings'

Reply:
happy to take consistency as a fast follow issue, just want to stem the tide of merge conflicts coming in

Reply:
OK thanks, yeah I think consistency across flavors is a big plus but it can happen in a follow-on